test/BrainPassValidator.t.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.13;
+
+import {PRBTest} from "prb-test/PRBTest.sol";
+import {Cheats} from "forge-std/Cheats.sol";
+import "forge-std/console.sol";
+import {stdError} from "forge-std/Errors.sol";
+import {BrainPassCollectibles} from "../src/BrainPass/BrainPass.sol";
+import {ERC721TokenReceiver} from "solmate/tokens/ERC721.sol";
+import {MockERC20} from "../lib/solmate/src/test/utils/mocks/MockERC20.sol";
+import {BrainPassValidiator} from "../src/BrainPass/BrainPassValidator.sol";
+
+contract BrainPassValidatorTest is PRBTest, Cheats {
+    BrainPassCollectibles BrainPass;
+    BrainPassValidiator brainPassValidator;
+    address alice = vm.addr(0x2);
+    address bob = vm.addr(0x3);
+    MockERC20 mockERC20;
+
+    function setUp() public {
+        mockERC20 = new MockERC20("Mock IQ Token", "MIT", 18); //mocking IQ token
+        BrainPass = new BrainPassCollectibles(address(mockERC20));
+        brainPassValidator = new BrainPassValidiator(address(BrainPass));
+        BrainPass.addPassType(15e18, "http://example.com", "Gold", 200, 0);
+    }
+
+    function testPostWiki() public {
```

Comment:
please do multiple tests for multiple use cases. there are at least 3 or 4 different cases